google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -0,0 +1,131 @@
+
+  private static final Logger logger =
+      Logger.getLogger(BigtableCloudMonitoringExporter.class.getName());
+  private final MetricServiceClient client;
+
+  private final String projectId;
+  private final String taskId;
+  private final MonitoredResource monitoredResource;
+  private boolean isShutdown = false;
+
+  private static final String RESOURCE_TYPE = "bigtable_client_raw";
+
+  static BigtableCloudMonitoringExporter create(String projectId, Credentials credentials)
+      throws IOException {
+    MetricServiceSettings.Builder settingsBuilder = MetricServiceSettings.newBuilder();
+    settingsBuilder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+
+    org.threeten.bp.Duration timeout = Duration.ofMinutes(1);
+    settingsBuilder.createServiceTimeSeriesSettings().setSimpleTimeoutNoRetries(timeout);
+    return new BigtableCloudMonitoringExporter(
+        projectId,
+        MetricServiceClient.create(settingsBuilder.build()),
+        MonitoredResource.newBuilder().setType(RESOURCE_TYPE).build(),
+        BigtableExporterUtils.getDefaultTaskValue());
+  }
+
+  @VisibleForTesting
+  BigtableCloudMonitoringExporter(
+      String projectId,
+      MetricServiceClient client,
+      MonitoredResource monitoredResource,
+      String taskId) {
+    this.client = client;
+    this.monitoredResource = monitoredResource;
+    this.taskId = taskId;
+    this.projectId = projectId;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> collection) {
+    if (isShutdown) {
+      return CompletableResultCode.ofFailure();
+    }
+    Preconditions.checkArgument(
+        collection.stream()
+            .flatMap(metricData -> metricData.getData().getPoints().stream())
+            .allMatch(pd -> BigtableExporterUtils.getProjectId(pd).equals(projectId)),
+        "Some metric data has unexpected projectId");
+    for (MetricData metricData : collection) {
+      if (!metricData.getInstrumentationScopeInfo().getName().equals("bigtable.googleapis.com")) {
+        continue;
+      }
+      List<TimeSeries> timeSeries =
+          metricData.getData().getPoints().stream()
+              .map(
+                  pointData ->
+                      BigtableExporterUtils.convertPointToTimeSeries(
+                          metricData, pointData, taskId, monitoredResource))
+              .collect(Collectors.toList());
+
+      ProjectName projectName = ProjectName.of(projectId);
+      CreateTimeSeriesRequest request =
+          CreateTimeSeriesRequest.newBuilder()
+              .setName(projectName.toString())
+              .addAllTimeSeries(timeSeries)
+              .build();
+
+      this.client.createServiceTimeSeriesCallable().futureCall(request);
+    }
+
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
```

Comment:
does this need to flush? I would imagine that you'd want to: set a closed flag so that new export calls are rejected, then flush, then return flush's CompletableResultCode